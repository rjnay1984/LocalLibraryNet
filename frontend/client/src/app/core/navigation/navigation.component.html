<mat-toolbar>
  <span>Local Library</span>
  <span class="example-spacer"></span>
  <button mat-icon-button class="example-icon person-icon" aria-label="Example icon-button with person icon">
    <mat-icon>person</mat-icon>
  </button>
  <span>Welcome User!</span>
</mat-toolbar>