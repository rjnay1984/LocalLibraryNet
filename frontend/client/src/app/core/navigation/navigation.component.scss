.mat-toolbar {
  position: fixed;
}

.example-spacer {
  flex: 1 1 auto;
}
