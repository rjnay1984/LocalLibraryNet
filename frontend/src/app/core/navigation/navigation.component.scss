.title-link {
  color: white;
  text-decoration: none;
}

.example-spacer {
  flex: 1 1 auto;
}

.navigation {
  display: flex;

  .navigation-list {
    a {
      color: white;
      text-decoration: none;

      &.active {
        text-decoration: underline;
      }
    }

    &:first-child {
      a {
        margin-right: 1em;
      }
    }
  }
}
