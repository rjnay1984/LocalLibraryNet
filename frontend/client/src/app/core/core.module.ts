import { CommonModule } from '@angular/common';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { NgModule } from '@angular/core';

import { MaterialModule } from '../shared/material.module';
import { ErrorInterceptor } from './interceptors/error.interceptor';
import { NavigationComponent } from './navigation/navigation.component';



@NgModule({
  declarations: [NavigationComponent],
  imports: [
    CommonModule,
    MaterialModule,
    HttpClientModule
  ],
  exports: [NavigationComponent],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: ErrorInterceptor, multi: true },
  ]
})
export class CoreModule { }
