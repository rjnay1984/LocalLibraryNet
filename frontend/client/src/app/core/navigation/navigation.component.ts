import { Component, OnInit } from '@angular/core';
import { ThemePalette } from '@angular/material/core';
import { AccountService } from '../services/account.service';

@Component({
  selector: 'app-navigation',
  templateUrl: './navigation.component.html',
  styleUrls: ['./navigation.component.scss'],
})
export class NavigationComponent implements OnInit {
  links = ['Login', 'Register'];
  activeLink = this.links[0];
  background: ThemePalette = undefined;

  constructor(public accountService: AccountService) {}

  ngOnInit(): void {}

  logout() {
    this.accountService.logout();
  }
}
