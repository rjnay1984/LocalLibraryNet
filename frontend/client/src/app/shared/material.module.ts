import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatToolbarModule } from '@angular/material/toolbar';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    MatToolbarModule,
    MatIconModule,
    MatButtonModule,
    MatFormFieldModule,
    MatInputModule,
    BrowserAnimationsModule
  ],
  exports: [
    MatButtonModule,
    MatIconModule,
    MatToolbarModule,
    MatFormFieldModule,
    MatInputModule,
    BrowserAnimationsModule
  ],
})
export class MaterialModule { }
