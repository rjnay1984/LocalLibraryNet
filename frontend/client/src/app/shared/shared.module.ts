import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CoreModule } from '../core/core.module';
import { LoginComponent } from './account/login/login.component';
import { MaterialModule } from './material.module';
import { ReactiveFormsModule } from '@angular/forms';



@NgModule({
  declarations: [LoginComponent],
  imports: [
    CommonModule,
    CoreModule,
    MaterialModule,
    ReactiveFormsModule
  ],
  exports: [LoginComponent]
})
export class SharedModule { }
