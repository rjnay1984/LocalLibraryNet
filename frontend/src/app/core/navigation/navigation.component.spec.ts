import { HttpClientModule } from '@angular/common/http';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatMenuModule } from '@angular/material/menu';
import { RouterTestingModule } from '@angular/router/testing';
import { MaterialModule } from 'src/app/shared/material.module';
import { SharedModule } from 'src/app/shared/shared.module';

import { NavigationComponent } from './navigation.component';

describe('NavigationComponent', () => {
  let component: NavigationComponent;
  let fixture: ComponentFixture<NavigationComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HttpClientModule, RouterTestingModule, MaterialModule],
      declarations: [NavigationComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(NavigationComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
