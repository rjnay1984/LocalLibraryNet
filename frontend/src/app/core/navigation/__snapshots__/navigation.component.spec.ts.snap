// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationComponent should show email when currentUser$ 1`] = `
<app-navigation
  accountService={[Function AccountService]}
  activeLink={[Function String]}
  background="undefined"
  links={[Function Array]}
>
  <mat-toolbar
    class="mat-elevation-z2 mat-toolbar mat-primary mat-toolbar-single-row"
    color="primary"
  >
    <a
      class="title-link"
      href="/"
    >
       Local Library 
    </a>
    <span
      class="example-spacer"
    />
    
    
    <button
      aria-haspopup="true"
      class="mat-focus-indicator mat-menu-trigger mat-button mat-button-base"
      mat-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          person
        </mat-icon>
         Welcome demouser@microsoft.com 
      </span>
      <span
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <span
        class="mat-button-focus-overlay"
      />
    </button>
    <mat-menu>
      
    </mat-menu>
    
  </mat-toolbar>
</app-navigation>
`;

exports[`NavigationComponent should show login/register when no currentUser$ 1`] = `
<app-navigation
  accountService={[Function AccountService]}
  activeLink={[Function String]}
  background="undefined"
  links={[Function Array]}
>
  <mat-toolbar
    class="mat-elevation-z2 mat-toolbar mat-primary mat-toolbar-single-row"
    color="primary"
  >
    <a
      class="title-link"
      href="/"
    >
       Local Library 
    </a>
    <span
      class="example-spacer"
    />
    
    <mat-nav-list
      class="navigation mat-nav-list mat-list-base"
      role="navigation"
    >
      <mat-list-item
        class="navigation-list mat-list-item mat-focus-indicator"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          >
            <a
              class="mat-line"
              href="/login"
              matline=""
              routerlinkactive="active"
            >
              Login
            </a>
          </div>
        </div>
      </mat-list-item>
      <mat-list-item
        class="navigation-list mat-list-item mat-focus-indicator"
      >
        <div
          class="mat-list-item-content"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          <div
            class="mat-list-text"
          >
            <a
              class="mat-line"
              href="/register"
              matline=""
              routerlinkactive="active"
            >
              Register
            </a>
          </div>
        </div>
      </mat-list-item>
    </mat-nav-list>
    
  </mat-toolbar>
</app-navigation>
`;
